src/main/scala/is/hail/linalg/LocalMatrix.scala

```diff
@@ -84,14 +81,32 @@ object LocalMatrix {
   }
 
   object ops {
-    implicit class ScalarShim(l: Double) {
-      def +(r: M): M = LocalMatrix(l +:+ r.m)
+    implicit class Shim(l: M) {
+      def +(r: M): M = l.add(r)
+
+      def -(r: M): M = l.subtract(r)
+
+      def *(r: M): M = l.multiply(r)
+
+      def /(r: M): M = l.divide(r)
+      
+      def +(r: Double): M = l.add(r)
 
-      def -(r: M): M = LocalMatrix(l -:- r.m)
+      def -(r: Double): M = l.subtract(r)
 
-      def *(r: M): M = LocalMatrix(l *:* r.m)
+      def *(r: Double): M = l.multiply(r)
 
-      def /(r: M): M = LocalMatrix(l /:/ r.m)
+      def /(r: Double): M = l.divide(r)   
+    }
+    
+    implicit class ScalarShim(l: Double) {      
+      def +(r: M): M = r + l
```

Comment:
I have a slight preference for these to call `r.add` and `r.multiply` directly to match `-` and `/`, but it is very slight and totally up to you.